### PR TITLE
Added support for NuGet update command.

### LINKED ISSUE #219: Add support for NuGet update command.

### PR DESCRIPTION
**Pull request by patriksvensson**
Should resolve #219 when (if) merged.

### CODE REVIEW

File: src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to look for updates with the highest 
+        /// version available within the same major and minor version as the installed package.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if safe; otherwise, <c>false</c>.
+        /// </value>
+        public bool Safe { get; set; }
+
```
Comment (devlead) [inline src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs]: Add optional `ICollection<string>` `Source` property to be able to specify alternative sources for update.
Like the `NugetRestoreSettings` i.e.
https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/NuGet/Restore/NugetRestoreSettings.cs#L32

File: src/Cake.Common/Tools/NuGet/Update/NuGetUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            builder.Append("update");
+            builder.AppendQuoted(targetFile.MakeAbsolute(_environment).FullPath);
+
+            // Packages?
+            if (settings.Id != null && settings.Id.Count > 0)
+            {
+                builder.Append("-Id");
+                builder.AppendQuoted(string.Join(";", settings.Id));
+            }
+
```
Comment (devlead) [inline src/Cake.Common/Tools/NuGet/Update/NuGetUpdater.cs]: Add optional Source parameter arguments
```csharp
            // List of package sources
            if (settings.Source != null && settings.Source.Count > 0)
            {
                builder.Append("-Source");
                builder.AppendQuoted(string.Join(";", settings.Source));
            }
```

**Comment by devlead** [comment]
Looks good! Some minor feedback above.

**Comment by patriksvensson** [comment]
Fixed :smile:

File: src/Cake.Common/Tools/NuGet/Update/NuGetUpdater.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuGetUpdater"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="nugetToolResolver">The nuget tool resolver.</param>
+        public NuGetUpdater(IFileSystem fileSystem, 
+            ICakeEnvironment environment,
+            IProcessRunner processRunner,
```
Comment (devlead) [inline src/Cake.Common/Tools/NuGet/Update/NuGetUpdater.cs]: Trailing space I'll fix in merge.

File: src/Cake.Common/Tools/NuGet/Update/NuGetUpdater.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        private readonly IToolResolver _nugetToolResolver;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuGetUpdater"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="nugetToolResolver">The nuget tool resolver.</param>
+        public NuGetUpdater(IFileSystem fileSystem,
```
Comment (devlead) [inline src/Cake.Common/Tools/NuGet/Update/NuGetUpdater.cs]: Trailing space I'll fix in merge.

File: src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        /// <value>The package ids to update.</value>
+        public ICollection<string> Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets a list of package sources to use for this command.
+        /// </summary>
+        public ICollection<string> Source { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to look for updates with the highest
```
Comment (devlead) [inline src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs]: Trailing space I'll fix in merge.

File: src/Cake.Common.Tests/Fixtures/NuGetUpdateFixture.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            Environment.WorkingDirectory.Returns("/Working");
+
+            Process = Substitute.For<IProcess>();
+            Process.GetExitCode().Returns(0);
+            ProcessRunner = Substitute.For<IProcessRunner>();
+            ProcessRunner.Start(Arg.Any<FilePath>(), Arg.Any<ProcessSettings>()).Returns(Process);
+
+            NuGetToolResolver = Substitute.For<IToolResolver>();
+            NuGetToolResolver.Name.Returns("NuGet");
+            NuGetToolResolver.ResolveToolPath().Returns(defaultToolExist
```
Comment (devlead) [inline src/Cake.Common.Tests/Fixtures/NuGetUpdateFixture.cs]: Trailing space I'll fix in merge.

**Pull request merged**